### PR TITLE
parca-agent: init at 0.35.0

### PR DESCRIPTION
**Pull request by jnsgruk**
Follow up to #359635 which introduces the agent component - an eBPF-based, always on profiling agent.

Tested locally against a bunch of different targets, uploading to the Polar Signals cloud and all seems to work nicely.

Also adds the `parca-debuginfo` companion tool at v0.11.0.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pa/parca-agent/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}:
+
+buildGoModule rec {
+  pname = "parca-agent";
+  version = "0.35.0";
+
+  src = fetchFromGitHub {
+    owner = "parca-dev";
+    repo = "parca-agent";
+    rev = "v${version}";
```
[the remark this answers is not in the document]
Comment (jnsgruk) [reply pkgs/by-name/pa/parca-agent/package.nix]: Fixed, thanks :)

File: pkgs/by-name/pa/parca-agent/package.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  ldflags = [
+    "-X=main.version=${version}"
+    "-X=main.commit=${src.rev}"
+    "-extldflags=-static"
+  ];
+
+  tags = [
+    "osusergo"
+    "netgo"
+  ];
```
Comment (drupol) [inline pkgs/by-name/pa/parca-agent/package.nix:36]: What is this?

Comment (jnsgruk) [reply pkgs/by-name/pa/parca-agent/package.nix:36]: Go build tags that need to be specified: https://nixos.org/manual/nixpkgs/stable/#var-go-tags

Background on the `netgo` tag here: https://pkg.go.dev/net#hdr-Name_Resolution, and the `osusergo` tag here: https://pkg.go.dev/os/user#pkg-overview

Specified in the upstream Makefile: https://github.com/parca-dev/parca-agent/blob/d59ef5c9bf21d7b422071c16ae83e01bb5fbcf2b/Makefile#L17

**Comment by jnsgruk** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360132`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>parca-agent</li>
    <li>parca-debuginfo</li>
  </ul>
</details>

**Pull request merged**